apps/emqx_authn/test/emqx_authn_pgsql_tls_SUITE.erl

```diff
@@ -0,0 +1,156 @@
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_authn_pgsql_tls_SUITE).
+
+-compile(nowarn_export_all).
+-compile(export_all).
+
+-include("emqx_authn.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("common_test/include/ct.hrl").
+
+-define(PGSQL_HOST, "pgsql-tls").
+-define(PGSQL_PORT, 5432).
+
+-define(PATH, [authentication]).
+
+all() ->
+    emqx_common_test_helpers:all(?MODULE).
+
+groups() ->
+    [].
+
+init_per_testcase(_, Config) ->
+    {ok, _} = emqx_cluster_rpc:start_link(node(), emqx_cluster_rpc, 1000),
+    emqx_authentication:initialize_authentication(?GLOBAL, []),
+    emqx_authn_test_lib:delete_authenticators(
+      [authentication],
+      ?GLOBAL),
+    Config.
+
+init_per_suite(Config) ->
+    _ = application:load(emqx_conf),
+    case emqx_authn_test_lib:is_tcp_server_available(?PGSQL_HOST, ?PGSQL_PORT) of
+        true ->
+            ok = emqx_common_test_helpers:start_apps([emqx_authn]),
+            ok = start_apps([emqx_resource, emqx_connector]),
+            Config;
+        false ->
+            {skip, no_pgsql_tls}
+    end.
+
+end_per_suite(_Config) ->
+    emqx_authn_test_lib:delete_authenticators(
+      [authentication],
+      ?GLOBAL),
+    ok = stop_apps([emqx_resource, emqx_connector]),
+    ok = emqx_common_test_helpers:stop_apps([emqx_authn]).
+
+%%------------------------------------------------------------------------------
+%% Tests
+%%------------------------------------------------------------------------------
+
+t_create(_Config) ->
+    %% openssl s_client -tls1_2 -cipher ECDHE-RSA-AES256-GCM-SHA384 \
+    %% -starttls postgres -connect pgsql-tls:5432 \
+    %% -cert pgsql-tls-client.crt -key pgsql-tls-client.key -CAfile pgsql-tls-ca.crt
+    ?assertMatch(
+       {ok, _},
+       create_pgsql_auth_with_ssl_opts(
+         #{<<"server_name_indication">> => <<"pgsql-tls">>,
+           <<"verify">> => <<"verify_peer">>,
+           <<"versions">> => [<<"tlsv1.2">>],
+           <<"ciphers">> => [<<"ECDHE-RSA-AES256-GCM-SHA384">>]})).
+
+t_create_invalid(_Config) ->
+
+    %% invalid server_name
+    ?assertMatch(
+       {error, _},
+       create_pgsql_auth_with_ssl_opts(
+         #{<<"server_name_indication">> => <<"pgsql-tls-unknown-host">>,
+           <<"verify">> => <<"verify_peer">>})),
+
+    %% invalid server_name 
+    ?assertMatch(
+       {error, _},
+       create_pgsql_auth_with_ssl_opts(
+         #{<<"verify">> => <<"verify_peer">>})),
```

Comment:
Apparently, this fails because the `"server_name_indication"` field is missing in the config.  Would this case also apply to MySQL as well?

Reply:
I removed this case.
We better not to test/rely on the fact that driver passes (or does not pass) `Host` directly to `ssl:connect`.